### PR TITLE
Tool Optimization

### PR DESCRIPTION
**Pull request by RoyalSix**
#### Describe what your pull request addresses (Please include relevant issue numbers):
- This PR has some optimizations for the tool opening 

#### Please include detailed Test instructions for your pull request:
- Checkout my branch on wA `feature-jay-5111`
- Try opening a fresh Acts project on develop and record the time.
- Try opening a fresh Acts project on this branch and record the time, it should be faster
.
#### Standard Test Instructions for PR Review Process:

- [ ] Double check unit tests that have been written
- [ ] Check for documentation for code changes
- [ ] Check that there are not inadvertent commits to tC Apps when reviewing a tC Core PR
- [ ] Checkout the branch locally and ensure that app runs as expected
  - [ ] Ensure tests pass
  - [ ] Open and watch the console for errors
  - [ ] Make sure all actions perform as expected
  - [ ] Import and Load a new Project
  - [ ] Load a tool and perform basic actions
  - [ ] Switch tools and perform basic actions
  - [ ] Switch project to an existing project
  - [ ] Load a tool and perform basic actions
  - [ ] Switch tools and perform basic actions
  - [ ] Next time reverse the order of importing after loading an existing project
- [ ] Reviewer should double check the DoD in the ISSUE, including the “spirit” of the story
- [ ] Ask Ben or Koz if you have any concerns about implementation (especially UI related)

---
This change is [Reviewable](https://reviewable.io/reviews/unfoldingword-dev/translationcore/5127)

### CODE REVIEW

**Comment by dsm-git** [comment]
[Linux build 1.0.0 (c44004d)](https://tc-builds.door43.org/build/feature-jay-5111/translationCore-linux-x64-1.0.0-c44004d.deb) is ready for download.
[macOS build 1.0.0 (c44004d)](https://tc-builds.door43.org/build/feature-jay-5111/translationCore-macos-x64-1.0.0-c44004d.dmg) is ready for download.
[Windows x64 build 1.0.0 (c44004d)](https://tc-builds.door43.org/build/feature-jay-5111/translationCore-win-x64-1.0.0-c44004d.setup.exe) is ready for download.
[Windows x32 build 1.0.0 (c44004d)](https://tc-builds.door43.org/build/feature-jay-5111/translationCore-win-x32-1.0.0-c44004d.setup.exe) is ready for download.

**Comment by dsm-git** [comment]
[Linux build 1.0.0 (3224286)](https://tc-builds.door43.org/build/feature-jay-5111/translationCore-linux-x64-1.0.0-3224286.deb) is ready for download.
[macOS build 1.0.0 (3224286)](https://tc-builds.door43.org/build/feature-jay-5111/translationCore-macos-x64-1.0.0-3224286.dmg) is ready for download.
[Windows x64 build 1.0.0 (3224286)](https://tc-builds.door43.org/build/feature-jay-5111/translationCore-win-x64-1.0.0-3224286.setup.exe) is ready for download.
[Windows x32 build 1.0.0 (3224286)](https://tc-builds.door43.org/build/feature-jay-5111/translationCore-win-x32-1.0.0-3224286.setup.exe) is ready for download.

File: gulpfile.js [hunk 1]
```diff
@@ -8,6 +8,22 @@ const change = require('gulp-change');
 const path = require('path');
 const rimraf = require('rimraf');
 const CrowdinApi = require('./scripts/CrowdinApi');
+const ncp = require('ncp').ncp;
+
```
Comment (mannycolon) [inline gulpfile.js]: @RoyalSix Hey any specific reason why you're editing the gulp file?

File: package.json [hunk 2]
```diff
@@ -132,6 +132,7 @@
     "json-stringify-safe": "5.0.1",
     "konami-code-js": "0.8.0",
     "lodash": "4.17.4",
+    "lodash.throttle": "^4.1.1",
```
Comment (mannycolon) [inline package.json]: :+1:

File: scripts/deb/DEBIAN/postinst [hunk 3]
```diff
@@ -0,0 +1,12 @@
+#!/bin/bash
+
+# install desktop launcher
+xdg-desktop-menu install /opt/translationcore/unfoldingword-translationcore.desktop
+
+# install icon
+xdg-icon-resource install --size 128 /opt/translationcore/translationcore-128.png unfoldingword-translationcore
```
Comment (mannycolon) [inline scripts/deb/DEBIAN/postinst]: @RoyalSix You have a bunch of new files that make me think you branch off of develop branch. is this correct?

**Review by mannycolon — CHANGES_REQUESTED** [review]

**Comment by mannycolon** [comment]
# Tool loading time:

## Acts project
  - `wA tool:` 17sec
  - `tW tool:` 22sec

## Titus project
  - `wA tool:` 7sec
  - `tW tool:` 8sec

**Pull request merged**